pkg/cmd/server/config/config.go

```diff
@@ -0,0 +1,263 @@
+package config
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/pflag"
+
+	"github.com/vmware-tanzu/velero/pkg/cmd/util/flag"
+	"github.com/vmware-tanzu/velero/pkg/podvolume"
+	"github.com/vmware-tanzu/velero/pkg/repository"
+	"github.com/vmware-tanzu/velero/pkg/uploader"
+	"github.com/vmware-tanzu/velero/pkg/util/logging"
+)
+
+const (
+	// the port where prometheus metrics are exposed
+	defaultMetricsAddress = ":8085"
+
+	defaultBackupSyncPeriod           = time.Minute
+	defaultStoreValidationFrequency   = time.Minute
+	defaultPodVolumeOperationTimeout  = 240 * time.Minute
+	defaultResourceTerminatingTimeout = 10 * time.Minute
+
+	// server's client default qps and burst
+	defaultClientQPS      float32 = 100.0
+	defaultClientBurst    int     = 100
+	defaultClientPageSize int     = 500
+
+	defaultProfilerAddress = "localhost:6060"
+
+	// the default TTL for a backup
+	defaultBackupTTL = 30 * 24 * time.Hour
+
+	defaultCSISnapshotTimeout   = 10 * time.Minute
+	defaultItemOperationTimeout = 4 * time.Hour
+
+	resourceTimeout = 10 * time.Minute
+
+	defaultMaxConcurrentK8SConnections = 30
+	defaultDisableInformerCache        = false
+
+	// defaultCredentialsDirectory is the path on disk where credential
+	// files will be written to
+	defaultCredentialsDirectory = "/tmp/credentials"
+
+	ControllerBackup                = "backup"
```

Comment:
We should consider decoupling the `pkg/cmd` and `pkg/controller` packages.
Maybe adding a `pkg/types` package to hold the common-used constant variables and structures can help to avoid packages importing each other.

Reply:
+1